astropy/coordinates/representation.py

```diff
@@ -1972,16 +1972,16 @@ def __init__(self, phi, theta=None, r=None, differentials=None, copy=True):
         super().__init__(phi, theta, r, copy=copy, differentials=differentials)
 
         # Wrap/validate phi/theta
-        if copy:
-            self._phi = self._phi.wrap_at(360 * u.deg)
-        else:
-            # necessary because the above version of `wrap_at` has to be a copy
-            self._phi.wrap_at(360 * u.deg, inplace=True)
-
-        if np.any(self._theta < 0.*u.deg) or np.any(self._theta > 180.*u.deg):
-            raise ValueError('Inclination angle(s) must be within '
-                             '0 deg <= angle <= 180 deg, '
-                             'got {}'.format(theta.to(u.degree)))
+        # Note that _phi already holds our own copy if copy=True.
+        self._phi.wrap_at(360 * u.deg, inplace=True)
```

Comment:
This is out of scope for this PR, but wondering now why we have to do this? Even in `SphericalRepresentation`, the longitude value gets wrapped by default because it gets passed into `Longitude`, which has no way to disable wrapping. But related to our discussions about internal representation of coordinate data and how we display it in frames, I wonder if we really need to wrap the data on input?

Reply:
Indeed, a good point, and it is a bit strange that we enforce the wrap angle on the user. On the other hand, for the latitude we do decide that it is between 0 and 180 (here) or 90 and -90 (for spherical). And if one converts from `Cartesian` one does get these specific ranges too.  Anyway, my sense is to let this be for now (and certainly for this PR!).